backend/core/llm_chain.py

```
@@ -3,490 +3,172 @@
-        # Use SHA-256 for secure, deterministic hashing
-        cache_hash = hashlib.sha256(cache_json.encode('utf-8')).hexdigest()
-
-        # Return shorter hash for efficiency (first 16 chars is plenty for collision avoidance)
-        return cache_hash[:16]
-
-    except Exception as e:
-        logger.error(f"Error generating cache key: {e}")
-        return None
-
+def get_cache_key(user_input: str) -> str:
+    if not ENABLE_CACHING or not isinstance(user_input, str): return None
```

Ruff (0.12.2)

Multiple statements on one line (colon)

(E701)